pkg/apim/APIMUtility.go

```diff
@@ -51,3 +66,21 @@ func (apim *Bundle) IsPayloadJSON() bool {
 	var js json.RawMessage
 	return json.Unmarshal([]byte(apim.Payload), &js) == nil
 }
+
+func (odataFilters *OdataParameters) InitOdataQuery() (string, error) {
+
+	v := reflect.ValueOf(odataFilters).Elem()
+	typeOfS := v.Type()
+	urlParam := url.Values{}
+	for i := 0; i < v.NumField(); i++ {
+		structVal := fmt.Sprintf("%v", v.Field(i).Interface())
+		structVal = strings.Replace(structVal, "[", "", 1)
+		structVal = strings.Replace(structVal, "]", "", 1)
+		if structVal != "" {
+			urlParam.Set(strings.ToLower(typeOfS.Field(i).Name), structVal)
```

Comment:
Could you not add the `$` here instead of the replace after `Encode`?

Reply:
it should be corrupted by encode call, so it is done later